platform/ios/src/MGLMapView.h

```diff
@@ -493,6 +493,19 @@ MGL_EXPORT IB_DESIGNABLE
  */
 @property(nonatomic, getter=isPitchEnabled) BOOL pitchEnabled;
 
+/**
+ A Boolean value that determines whether the user will receive haptic feedback
+ for certain interactions with the map.
+
+ When this property is set to `YES`, the default, a `UIImpactFeedbackStyleLight`
+ haptic feedback event be played when the user rotates the map to due north
+ (0°).
+
+ This feature requires a device that supports haptic feedback, running iOS 10 or
+ newer.
+ */
+@property(nonatomic, getter=isHapticFeedbackEnabled) BOOL hapticFeedbackEnabled;
```

Comment:
I wonder if we’ll eventually outgrow this single Boolean flag. Something akin to AppKit’s `NSHapticFeedbackPattern` enumeration would give us a bit more flexibility.

Reply:
Which would be the events that may provide haptic feedback?

Reply:
For now, just compass snapping, though https://github.com/mapbox/mapbox-gl-native/pull/10847#discussion_r160525274 hints at a couple more possibilities. An enumeration with “haptic feedback levels” would give us room to add more in the future without overwhelming the default behavior.

Reply:
Having an enum would allow us to offer a higher/broader level of feedback as a configurable option, which could be nice. I think that’s a `#future` thing, though, when/if we’re able to find multiple other use cases for rumbling people’s devices.

![nintendo_rumble_pak](https://user-images.githubusercontent.com/1198851/34803362-e527ad70-f640-11e7-936c-8c321291f1f1.jpg)

Reply:
Understood, though note that this design locks us into a binary option until v5.0.0.

Reply:
That's a very good point. An enum with just two values (on and off, essentially) would make this easier to extend in the `#future` without the need for a deprecation and subsequent breaking change.

Reply:
What do we call the `on` enum value if we don’t know what future options will be? 🤔

Reply:
I was thinking of a bitfield along the lines of [`MGLDebugMaskOptions`](https://github.com/mapbox/mapbox-gl-native/blob/b2aa8d6ca5d1b6eb7b6e52d817735eabea9d8de0/platform/darwin/src/MGLTypes.h#L54-L78), except perhaps with an `All` option that occupies the entire width of the type.